### PR TITLE
Order date labels chronologically for CPI.

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

Order date labels chronologically for CPI.

Date formats are currently ordered alphabetically. This change allows
formats to be recognised and parsed allowing the dates to be sorted
chronologically.

### How to review

Review code + tests

### Who can review

Anyone

### CODE REVIEW

File: src/main/java/dp/xlsx/DateLabel.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Helper for working with date labels in v4 files.
+ */
+public class DateLabel {
+
+    private static final Map<String, String> DATE_FORMAT_REGEXPS = new HashMap<String, String>() {{
+        put("^\\w{3}-\\d{2}$", "MMM-yy");
```
Comment (daiLlew) [inline src/main/java/dp/xlsx/DateLabel.java]: I've never seen this syntax before - no idea you could even do this.

File: src/main/java/dp/xlsx/DateLabel.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    }};
+
+    /**
+     * Take a date label and return it's format if it's recognised.
+     */
+    public static String determineDateFormat(String dateString) {
+
+        for (String regexp : DATE_FORMAT_REGEXPS.keySet()) {
+            if (dateString.toLowerCase().matches(regexp)) {
+                return DATE_FORMAT_REGEXPS.get(regexp);
```
Comment (daiLlew) [inline src/main/java/dp/xlsx/DateLabel.java]: We only have one thing in there at the moment but in the future we might need to consider the case where more than one regex matches - if we get a hit on the wrong one but thats not an issue for now.

File: src/main/java/dp/xlsx/DateLabel.java (hunk 1 above)
Comment (CarlHembrough) [reply src/main/java/dp/xlsx/DateLabel.java]: 👍

File: src/main/java/dp/xlsx/DateLabel.java (hunk 2 above)
Comment (CarlHembrough) [reply src/main/java/dp/xlsx/DateLabel.java]: 👍

**Pull request merged**